dart/lib/src/protocol/sentry_span.dart

```diff
@@ -183,6 +183,15 @@ class SentrySpan extends ISentrySpan {
         sampled: samplingDecision?.sampled,
       );
 
+  @override
+  void setMeasurement(
+    String name,
+    num value, {
+    SentryMeasurementUnit? unit,
+  }) {
```

Comment:
`h`: I think a nicer API would be to have multiple methods for different units. One for num, one for int, one for DateTime, etc. I know there is no method overloading in Dart, but there are optional types or something. If that's the dart way of doing it, ignore me, please.

Reply:
One way of doing that would be to just take a `SentryMeasurement` object and have a lot of different constructors on `SentryMeasurement`.

```dart
// pseudo code
class SentryMeasurement {
  factory SentryMeasurement.fromTimeSpan(TimeSpan timespan) {
    return SentryMeasurement(timeSpan.milliseconds); // or whatever makes sense
  }
  // more constructors
}

class SentrySpan {
  void setMeasurement(String name, SentryMeasurement measurement) {
    // implementation ...
  }
}
```

Reply:
The units are defined in the protocol.
This is the Dart overload already, which optionally takes a `unit` or defaults to `none` as mandated.
Otherwise, we'd need to add 8 different methods which is worse IMO.

Reply:
Yeah using named ctor helps, we already have some with the pre-defined measurements.
I have kept it as it is since its a Unified API across other SDKs.
Users can still make extension methods by converting a `Duration` to their preferred units.